networks/inference_net.py

```diff
@@ -0,0 +1,276 @@
+"""This module contains the implementation of the inference step of the Hamiltonian Generative
+Networks paper. The inference step is composed by an Encoder network and a Transformer network.
+The encoder maps the input sequence of frames into a latent distribution and samples a Tensor z
+from it using the reparametrization trick. The Transformer network takes the latent Tensor z and
+maps it into the abstract phase space (q, p).
+"""
+
+import torch
+import torch.nn as nn
+
+
+class EncoderNet(nn.Module):
+    """Implementation of the encoder network, that encodes the input frames sequence into a
+    distribution over the latent space and samples with the common reparametrization trick.
+
+    The network expects the images to be concatenated along channel dimension. This means that if
+    a batch of sequences has shape (batch_size, seq_len, channels, height, width) the network
+    will accept an input of shape (batch_size, seq_len * channels, height, width).
+    """
+
+    DEFAULT_PARAMS = {
+        'hidden_conv_layers': 6,
```

Comment:
Isn't that number implicit in the len of the lists?

Reply:
It is not really intuitive, as the `kernel_sizes` and `strides` lists have len equal to the total number of layers (`hidden_conv_layers + 2`),  while `n_filters` must have len equal to the total number of layers - 1, which is `hidden_conv_layers + 1`

Reply:
oh i see, ok then